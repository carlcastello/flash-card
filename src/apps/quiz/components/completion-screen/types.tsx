export interface IOwnProps {
  onClose: () => void,
  onReview: () => void,
  classes: {
    
  }
}