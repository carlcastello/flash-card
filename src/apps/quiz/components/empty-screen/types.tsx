export interface IOwnProps {
  onEdit: () => void,
  onClose: () => void,
  classes: {

  }
}