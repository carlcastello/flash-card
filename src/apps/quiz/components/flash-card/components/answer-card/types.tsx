import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  answer: string,
  flashcardStatus: FlashcardStatus,
  ref: any,
  classes: {
    paper: string,
    input: string,
    typography: string
  }
}

export interface IOwnState {
  answer: string,
  hideAnswer: boolean,
}