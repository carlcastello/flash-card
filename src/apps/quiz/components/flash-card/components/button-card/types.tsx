import { FlashcardStatus } from "../../types";

export interface IOwnProps {
  flashcardStatus: FlashcardStatus,
  submit: () => void,
  next: () => void,
  classes: {
    button: string
  }
}

export interface IOwnState {

}