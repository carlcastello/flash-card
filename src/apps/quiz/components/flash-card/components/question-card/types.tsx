import { FlashCardStatus } from "../../types";
import { QuestionType } from "../../../../../commons/types";

export interface IQuestionObject {
  question: string,
  subQuestion?: string,
  hint?: string,
  questionType: QuestionType,
}

export interface IOwnProps {
  flashCardStatus: FlashCardStatus,
  questionObject: IQuestionObject,
  onSkip: () => void,
  classes: {
    paper: string,
    boxButtonContainer: string,
    boxQuestionaireContainer: string
  }
}