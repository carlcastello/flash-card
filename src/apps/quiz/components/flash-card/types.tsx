import { IFlashCard } from "../../../commons/types";

export enum FlashCardStatus {
  DEFAULT = "DEFAULT",
  CORRECT = "CORRECT",
  WRONG = "WRONG",
  HINT = "HINT"
}

export interface IOwnProps {
  update: (flashCardStatus: FlashCardStatus) => void,
  next: () => void,
  flashCardObject: IFlashCard,
  classes: {

  }
}

export interface IOwnState {
  flashCardStatus: FlashCardStatus
}