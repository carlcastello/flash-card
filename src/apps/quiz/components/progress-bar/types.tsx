import { FlashcardStatus } from "../flash-card/types";

export interface IOwnProps {
  currentQuestion: number,
  totalQuestion: number,
  flashcardStatus: FlashcardStatus,
  classes: {
    title: string,
    progressBarContainer: string,
    progressBar: string,
  }
}