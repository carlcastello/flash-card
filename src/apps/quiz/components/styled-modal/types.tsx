export interface IOwnProps {
  isOpen: boolean,
  onConfirm?: () => void,
  onIgnore: () => void,

  classes: {
    modalContainer: string,
    modalPaperContainer: string,
    closeButton: string,
  }
}