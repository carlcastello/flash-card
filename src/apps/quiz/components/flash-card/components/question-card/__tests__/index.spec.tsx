import React from "react";
import { shallow } from "enzyme";

import { QuestionCard } from "../index";
import { FlashcardStatus } from "../../../types";
import toJson from "enzyme-to-json";
import { Typography, IconButton } from "@material-ui/core";

describe("", () => {

  let onSkipFunction: any;

  let wrapper: any;

  beforeEach(() => {
    onSkipFunction = jest.fn();

    wrapper = shallow(
      <QuestionCard
        flashcardStatus={FlashcardStatus.DEFAULT}
        questionObject={{
          question: "Potato Salad",
          hint: "This is a hint",
        }}
        onSkip={onSkipFunction}
        classes={{
          paper: "paper",
          boxButtonContainer: "box-button-container",
          boxQuestionaireContainer: "box-questionaire-container",
          questionTypography: "question-typography"
        }}/>
    );
  });

  describe("The render function", () => {
    it("renders as FlashcardStatus.DEFAULT with subQuestion", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Typography)).toHaveLength(1);
      expect(wrapper.find(IconButton).props().disabled).toEqual(false);
    })

    it("renders as none FlashcardStatus.DEFAULT without subQuestion", () => {
      wrapper.setProps({
        flashcardStatus: FlashcardStatus.HINT,
        questionObject: {
          question: "Potato Salad",
          hint: "This is a hint",
          subQuestion: "this is a sub"
        }
      });

      expect(toJson(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Typography)).toHaveLength(2);
      expect(wrapper.find(IconButton).props().disabled).toEqual(true);
    });
  });
});