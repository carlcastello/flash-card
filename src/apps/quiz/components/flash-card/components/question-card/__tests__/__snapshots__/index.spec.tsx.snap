// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` The render function renders as FlashcardStatus.DEFAULT with subQuestion 1`] = `
<WithStyles(ForwardRef(Paper))
  className="paper"
  elevation={3}
>
  <Styled(MuiBox)
    className="box-questionaire-container"
    px={2}
    py={4}
  >
    <WithStyles(ForwardRef(Typography))
      className="question-typography"
      variant="h3"
    >
      Potato Salad
    </WithStyles(ForwardRef(Typography))>
    <Styled(MuiBox)
      className="box-button-container"
      pb={1}
      pr={2}
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={false}
        onClick={[MockFunction]}
      >
        <SkipNextIcon />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;

exports[` The render function renders as none FlashcardStatus.DEFAULT without subQuestion 1`] = `
<WithStyles(ForwardRef(Paper))
  className="paper"
  elevation={3}
>
  <Styled(MuiBox)
    className="box-questionaire-container"
    px={2}
    py={4}
  >
    <WithStyles(ForwardRef(Typography))
      className="question-typography"
      variant="h3"
    >
      Potato Salad
    </WithStyles(ForwardRef(Typography))>
    <Styled(MuiBox)
      mt={0.5}
    >
      <WithStyles(ForwardRef(Typography))
        variant="body1"
      >
        this is a sub
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      className="box-button-container"
      pb={1}
      pr={2}
    >
      <WithStyles(ForwardRef(IconButton))
        disabled={true}
        onClick={[MockFunction]}
      >
        <SkipNextIcon />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;
